tracer/src/Datadog.Trace/Debugger/Instrumentation/MethodDebuggerInvoker.cs

```diff
@@ -36,7 +37,7 @@ public static class MethodDebuggerInvoker
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static MethodDebuggerState BeginMethod_StartMarker<TTarget>(string probeId, TTarget instance, RuntimeMethodHandle methodHandle, RuntimeTypeHandle typeHandle, int methodMetadataIndex)
         {
-            if (ProbeRateLimiter.Instance.IsLimitReached)
+            if (!ProbeRateLimiter.Instance.Sample(probeId))
             {
```

Comment:
Deciding against it for now since it seems like it might pollute our log files too much, but if we find that we really need it in the future, we can always reconsider.